test/integration/acceptance/remove_signatory_test.cpp

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+class RemoveSignatory : public AcceptanceFixture {
+ public:
+  auto makeFirstUser(const interface::RolePermissionSet &perms = {
+                         interface::permissions::Role::kRemoveSignatory}) {
+    auto new_perms = perms;
+    new_perms.set(interface::permissions::Role::kAddSignatory);
+    return AcceptanceFixture::makeUserWithPerms(new_perms);
+  }
+
+  auto makeSecondUser(const interface::RolePermissionSet &perms = {
+                          interface::permissions::Role::kReceive}) {
+    return complete(
+        createUserWithPerms(kUser2, kUser2Keypair.publicKey(), kRole2, perms)
+            .creatorAccountId(kAdminId),
+        kAdminKeypair);
+  }
+
+  const std::string kRole2 = "roletwo";
+  const std::string kUser2 = "usertwo";
+  const std::string kUser2Id = kUser2 + "@test";
+  const std::string kAmount = "1.0";
+  const crypto::Keypair kUser2Keypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+#define CHECK_BLOCK(i) \
+  [](auto &block) { ASSERT_EQ(block->transactions().size(), i); }
+
+/**
+ * C264 Remove signatory from own account
+ * C267 Remove signatory more than once
+ * @given some user with CanRemoveSignatory permission and its signatory
+ * @when execute tx with RemoveSignatory where the first is a creator and the
+ *       second's key is removed as a signatory
+ * @then the first such tx is committed,
+         the same transaction afterward shouldn't pass stateful validation
+ */
+TEST_F(RemoveSignatory, Basic) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(makeFirstUser(), CHECK_BLOCK(1))
+      .sendTxAwait(makeSecondUser(), CHECK_BLOCK(1))
+      .sendTxAwait(
+          complete(baseTx().addSignatory(kUserId, kUser2Keypair.publicKey())),
+          CHECK_BLOCK(1))
+      .sendTxAwait(complete(baseTx().removeSignatory(
+                       kUserId, kUser2Keypair.publicKey())),
+                   CHECK_BLOCK(1))
```

Comment:
Please split the test case into two.
The basic case (C264) should be as the existing up to the commented line.
The second case (C267) should be equal to the whole this test case.